plugins/nf-prov/src/main/nextflow/prov/WrrocRenderer.groovy

```diff
@@ -0,0 +1,960 @@
+                }
+            }
+        }
+
+        // get workflow config and store it in crate
+        final configFilePath = crateRootDir.resolve("nextflow.config")
+        final configFileWriter = new FileWriter(configFilePath.toString())
+        configMap.toConfigObject().writeTo(configFileWriter)
+
+        // get workflow README file and store it in crate
+        Map readmeFile = null
+
+        for( final fileName : README_FILENAMES ) {
+            final readmeFilePath = projectDir.resolve(fileName)
+            if( !Files.exists(readmeFilePath) )
+                continue
+
+            final encoding = FilenameUtils.getExtension(fileName).equals("md")
+                ? "text/markdown"
+                : "text/plain"
+            readmeFile = [
+                "@id"           : fileName,
+                "@type"         : "File",
+                "name"          : fileName,
+                "description"   : "This is the README file of the workflow.",
+                "encodingFormat": encoding
+            ]
+            Files.copy(readmeFilePath, crateRootDir.resolve(fileName), StandardCopyOption.REPLACE_EXISTING)
+            break
+        }
+
+        // get workflow metadata
+        final metadata = session.workflowMetadata
+        this.normalizer = new PathNormalizer(metadata)
+
+        final manifest = metadata.manifest
+        final nextflowMeta = metadata.nextflow
+        final scriptFile = metadata.getScriptFile()
+
+        final formatter = DateTimeFormatter.ISO_OFFSET_DATE_TIME
+        final dateStarted = formatter.format(metadata.start)
+        final dateCompleted = formatter.format(metadata.complete)
+        final nextflowVersion = nextflowMeta.version.toString()
+        final wrrocParams = session.config.navigate('prov.formats.wrroc', [:]) as Map
+
+        // Copy workflow into crate directory
+        Files.copy(scriptFile, crateRootDir.resolve(scriptFile.getFileName()), StandardCopyOption.REPLACE_EXISTING)
+
+        // Copy nextflow_schema_json into crate if it exists
+        final schemaFile = scriptFile.getParent().resolve("nextflow_schema.json")
+        // TODO Add to crate metadata
+        if( Files.exists(schemaFile) )
+            Files.copy(schemaFile, crateRootDir.resolve(schemaFile.getFileName()), StandardCopyOption.REPLACE_EXISTING)
+
+        // create manifest
+        final softwareApplicationId = UUID.randomUUID()
+        final organizeActionId = UUID.randomUUID()
+
+        // Process wrroc configuration options
+        agent = parseAgentInfo(wrrocParams)
+        organization = parseOrganizationInfo(wrrocParams)
+        publisherID = getPublisherID(wrrocParams, agent, organization)
+        if( organization )
+            agent.put("affiliation", ["@id": organization.get("@id")])
+
+        // license information
+        final license = [
+            "@id"  : manifest.license,
+            "@type": "CreativeWork"
+        ]
```

Comment:
Nextflow 24.10 added the `manifest.license` config option. Since the latest version of nf-prov requires 24.10 , I went ahead and refactored this bit to use this option instead of a custom option